From 5b05b822389abb8377bb79f61303f14155714229 Mon Sep 17 00:00:00 2001
From: Wallace Wang <r59996@freescale.com>
Date: Wed, 19 Mar 2008 14:27:06 +0800
Subject: [PATCH] ENGR00069205 CSR BT porting on i.MX31 3stack

Enable the uart2 for BT
Fix the mxc_gpio_request for none gpio pins

Signed-off-by: Wallace Wang <r59996@freescale.com>
---
 arch/arm/mach-mx3/board-mx3_3stack.h |    2 +-
 arch/arm/mach-mx3/mx3_3stack_gpio.c  |   10 ++++------
 2 files changed, 5 insertions(+), 7 deletions(-)

diff --git a/arch/arm/mach-mx3/board-mx3_3stack.h b/arch/arm/mach-mx3/board-mx3_3stack.h
index 78a1e92..bd98dc5 100644
--- a/arch/arm/mach-mx3/board-mx3_3stack.h
+++ b/arch/arm/mach-mx3/board-mx3_3stack.h
@@ -71,7 +71,7 @@
 /*! @} */
 /* UART 2 configuration */
 #define UART2_MODE              MODE_DCE
-#define UART2_IR                IRDA
+#define UART2_IR                NO_IRDA
 #define UART2_ENABLED           1
 /* UART 3 configuration */
 #define UART3_MODE              MODE_DTE
diff --git a/arch/arm/mach-mx3/mx3_3stack_gpio.c b/arch/arm/mach-mx3/mx3_3stack_gpio.c
index d99c6ac..eaf9e0f 100644
--- a/arch/arm/mach-mx3/mx3_3stack_gpio.c
+++ b/arch/arm/mach-mx3/mx3_3stack_gpio.c
@@ -112,17 +112,15 @@ void gpio_uart_inactive(int port, int no_irda)
 	case 1:
 		mxc_request_gpio(MX31_PIN_TXD2);
 		mxc_request_gpio(MX31_PIN_RXD2);
-		mxc_request_gpio(MX31_PIN_RTS2);
-		mxc_request_gpio(MX31_PIN_CTS2);
 
 		mxc_free_iomux(MX31_PIN_TXD2, OUTPUTCONFIG_GPIO,
 			       INPUTCONFIG_GPIO);
 		mxc_free_iomux(MX31_PIN_RXD2, OUTPUTCONFIG_GPIO,
 			       INPUTCONFIG_GPIO);
-		mxc_free_iomux(MX31_PIN_RTS2, OUTPUTCONFIG_GPIO,
-			       INPUTCONFIG_GPIO);
-		mxc_free_iomux(MX31_PIN_CTS2, OUTPUTCONFIG_GPIO,
-			       INPUTCONFIG_GPIO);
+		mxc_free_iomux(MX31_PIN_RTS2, OUTPUTCONFIG_FUNC,
+			       INPUTCONFIG_FUNC);
+		mxc_free_iomux(MX31_PIN_CTS2, OUTPUTCONFIG_FUNC,
+			       INPUTCONFIG_FUNC);
 		break;
 	default:
 		break;
-- 
1.5.4.4

